### PR TITLE
Add spelling workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
It will find misspellings.

### CODE REVIEW

**Comment by szepeviktor** [comment]
```
Warning: "Verison" should be "Version".
Warning: "Overriden" should be "Overridden".
Warning: "Removeable" should be "Removable".
```

- https://github.com/search?q=supportsLessOverridenParametersWithVariadic+NOT+is%3Afork&type=code
- https://github.com/search?q=NonRemoveableTypeTrait+NOT+is%3Afork&type=code

What should I correct?
Please see annotations on [the diff page](https://github.com/phpstan/phpstan-src/pull/3179/files).

**Comment by staabm** [comment]
Variables can be fixed. 

Methods not declared `@api` or in classes not declared `@api` can be fixed. 

I am not sure about `NonRemoveableTypeTrait`

File: src/Php/PhpVersion.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function deprecatesRequiredParameterAfterOptionalUnionOrMixed(): bool
 		return $this->versionId >= 80300;
 	}
 
-	public function supportsLessOverridenParametersWithVariadic(): bool
+	public function supportsLessOverriddenParametersWithVariadic(): bool
```
Comment (staabm) [inline src/Php/PhpVersion.php]: the PhpVersion class is declared `@api` therefore you cannot change this public method name here (BC break)

**Comment by szepeviktor** [comment]
I've fixes a method name in `PhpVersion` although it is marked `@api`.
Seems to be unused: https://github.com/search?q=supportsLessOverridenParametersWithVariadic+NOT+is%3Afork&type=code

File: src/Php/PhpVersion.php (hunk 1 above)
Comment (szepeviktor) [reply src/Php/PhpVersion.php]: I've fixes a method name in `PhpVersion` although it is marked `@api`.
Seems to be unused: https://github.com/search?q=supportsLessOverridenParametersWithVariadic+NOT+is%3Afork&type=code

Comment (szepeviktor) [reply src/Php/PhpVersion.php]: Should I revert it?

Comment (staabm) [reply src/Php/PhpVersion.php]: this method might be used in 3rd party extensions (thats what the `@api` is meant for)

Comment (szepeviktor) [reply src/Php/PhpVersion.php]: Reverted 🍏

**Comment by ondrejmirtes** [comment]
Thank you.

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**